tests/run/datetime_pxd.pyx

```diff
@@ -238,3 +239,31 @@ def test_datetime_attrs_inlined(datetime dt):
         dt.second,
         dt.microsecond,
     )
+
+def test_date_from_utc_timestamp():
+    """
+    >>> from datetime import datetime
+    >>> tp, dt = test_date_from_utc_timestamp()
+    >>> tp == dt
+    True
+    """
+    tp = date_from_timestamp(1518185542)
+    dt = py_datetime.date(2018, 2, 9)
+    return tp, dt
+
+def test_datetime_from_utc_timestamp():
+    """
+    >>> from datetime import datetime
+    >>> test_datetime_from_utc_timestamp()
+    True
+    """
+    try:
+        tp = datetime_from_timestamp(1618770890, get_utc())
+        dt = py_datetime.datetime(2021, 4, 18, 18, 34, 50, tzinfo=py_datetime.timezone.utc)
+        return tp == dt
+    except RuntimeError:
```

Comment:
This should be more explicit about what part is allowed to raise the exception. It's only `get_utc()` that we expect to fail, right? Then only that should be wrapped by try-except. Otherwise, we risk hiding bugs.